src/NaCl.Core/Base/Salsa20Base.cs

```diff
@@ -112,7 +90,7 @@ public void HSalsa20InitialState(Span<uint> state, ReadOnlySpan<byte> nonce)
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        protected static void ShuffleState(Span<uint> state)
+        internal protected static void ShuffleState(Span<uint> state)

```

Comment:
Something minor... the access modifier should be `protected internal`, just need to fix the ordering.